src/sources/file.rs

```diff
@@ -210,7 +210,8 @@ impl SourceConfig for FileConfig {
         // other
         let data_dir = cx
             .globals
-            .resolve_and_make_data_subdir(self.data_dir.as_ref(), &cx.id)?;
+            // source are only global, name can be used for subdir
+            .resolve_and_make_data_subdir(self.data_dir.as_ref(), &cx.id.to_string())?;
```

Comment:
`resolve_and_make_data_subdir` takes a &str; this could be satisfied with `&cx.id.id` in the current public context, or `cx.id.as_str()` per previous suggestions, and avoid the allocation.